DEP: Remove split `FutureWarning` for empty arrays in the result.

charris opened the pull request:
DEP: Remove FutureWarning from np.lib.split and go to future.

Previously an empty array resulting from split always had dimension 1-D.
In Numpy 1.9 a FutureWarning was raised to notify users that it was
planned to preserve the dimensions of empty arrays in a future numpy
release. This removes the FutureWarning and implements preservation of
dimensions.

Note that there was a bug in numpy 1.9 and the dimensions of empty
arrays was already preserved in some cases and no warning was issued.
This PR fixes that inconsistency by preserving the dimensions in all
cases rather than fixing the bug, as the dimension preserving behavior
was already depended on by some users. See the discussion in gh-6575
about this change.

charris committed:
DEP: Remove FutureWarning from np.lib.split and go to future.

Previously an empty array resulting from split always had dimension 1-D.
In Numpy 1.9 a FutureWarning was raised to notify users that it was
planned to preserve the dimensions of empty arrays in a future numpy
release. This removes the FutureWarning and implements preservation of
dimensions.

Note that there was a bug in numpy 1.9 and the dimensions of empty
arrays was already preserved in some cases and no warning was issued.
This PR fixes that inconsistency by preserving the dimensions in all
cases rather than fixing the bug, as the dimension preserving behavior
was already depended on by some users. See the discussion in gh-6575
about this change.

charris committed:
DOC: Update 1.10.0 release notes for future changes to np.lib.split.

Note that empty array resulting from a split will preserve dimensions
starting in Numpy 1.11.

njsmith commented:
Did the behavior for non-last arrays actually change between 1.8 and 1.9?
(I ask because if so, and no one complained, that would be more data in
favor of changing this on a somewhat aggressive timeline.)
On Oct 30, 2015 1:14 PM, "Charles Harris" <notifications@github.com> wrote:

> DEP: Remove FutureWarning from np.lib.split and go to future.
>
> An empty array resulting from split always had dimension 1-D but
> it was planned to make all resulting arrays preserve dimensions,
> hence the FutureWarning. This implements preservation of dimensions
> and the FutureWarning is removed.
>
> Note that there was actually a bug in numpy 1.9 and the dimension
> preserving behavior was already in place when the empty array was
> not the last of the arrays in the split and that was not warned about.
>
> See the discussion in gh-6575 <https://github.com/numpy/numpy/issues/6575>
> about this change.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/numpy/numpy/pull/6592
> Commit Summary
>
>    - DEP: Remove FutureWarning from np.lib.split and go to future.
>    - DOC: Update 1.10.0 release notes for future changes to np.lib.split.
>
> File Changes
>
>    - *M* doc/release/1.10.0-notes.rst
>    <https://github.com/numpy/numpy/pull/6592/files#diff-0> (16)
>    - *M* numpy/lib/shape_base.py
>    <https://github.com/numpy/numpy/pull/6592/files#diff-1> (11)
>    - *M* numpy/lib/tests/test_shape_base.py
>    <https://github.com/numpy/numpy/pull/6592/files#diff-2> (30)
>
> Patch Links:
>
>    - https://github.com/numpy/numpy/pull/6592.patch
>    - https://github.com/numpy/numpy/pull/6592.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/6592>.
>

charris commented:
Yes, and the fix for that in 1.10 was what led to #6575.

njsmith commented:
No objections here then, and the actual PR LGTM.

Might be good to send a pro forma notification to the mailing list, and wait a few days in case someone objects.

charris committed:
DOC: Update 1.11.0 release notes with np.lib.split changes.

Note that the FutureWarning has been removed and that empty arrays
resulting from np.lib.split will preserve dimensions instead of always
having shape (0,).
